docs/reference/esql/source-commands/from.asciidoc

```diff
@@ -2,10 +2,34 @@
 [[esql-from]]
 === `FROM`
 
-The `FROM` source command returns a table with up to 10,000 documents from a
-data stream, index, or alias. Each row in the resulting table represents a
-document. Each column corresponds to a field, and can be accessed by the name
-of that field.
+**Syntax**
+
+[source,esql]
+----
+FROM index_pattern [METADATA fields]
+----
+
+*Parameters*
+
+`index_pattern`::
+A list of indices, data streams or aliases. Supports wildcards and date math.
+
+`fields`::
+A comma-separated list of <<esql-metadata-fields,metadata fields>> to retrieve.
+
+*Description*
+
+The `FROM` source command returns a table with data from a data stream, index,
+or alias. Each row in the resulting table represents a document. Each column
+corresponds to a field, and can be accessed by the name of that field.
+
+By default, `FROM` retrieves up to 500 rows. Use <<esql-limit>> to set a another
```

Comment:
I have somewhat mixed feelings about this section. `FROM` in itself doesn't limit the number of rows, it's the `limit` command that does this. Using a query like `FROM test` will have by default a `limit 500` added to it, but this is slightly different than saying that `FROM retrieves up to 500 rows`. Imho, this section should be removed.

Reply:
I've checked the `limit` docs section and I believe it's spelled out clearly there what we do with these default numbers.
Maybe reword this one for `FROM` to something like: 
> By default, a query without an explicit <<esql-limit>> will return up to 500 rows. This also applies to `FROM` command, by piping a default `LIMIT 500` to a query like `FROM test`. Using <<esql-limit>> you can increase the number of rows up to 10,000. ES|QL queries will not return more than 10,000 rows, regardless of the `LIMIT` command's value. You can change the maximum number of rows with the `esql.query.result_truncation_max_size` static cluster setting."

Reply:
That's a good point. The implicit limit has nothing to do with `FROM`. It is worth calling out here though, as I think it's something users will run into when they write their first ES|QL query in Discover. How about we turn it into the following "NOTE" callout instead:

<img width="724" alt="image" src="https://github.com/elastic/elasticsearch/assets/9715543/d216bd93-07e2-44ca-ada5-8132790c34fb">

We can then leave the 10,000 max discussion to the LIMIT docs.

Reply:
Perfect 👍